### PR TITLE
Misuv 7493 exceptions thrown in view

### PR DESCRIPTION
**Pull request by james-king-hmrc**
- code refactor to target change: tech-debt task to remove code throwing exceptions from the view template file

### CODE REVIEW

File: app/controllers/ChargeSummaryController.scala [hunk 1]
```diff
@@ -40,6 +40,20 @@ import views.html.errorPages.CustomNotFoundError
...
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
+import scala.language.implicitConversions
+import controllers.ChargeSummaryController._
+
+object ChargeSummaryController {
+  case class ErrorCode(message: String, code: Int = 0, showInternalServerError: Boolean = true) extends RuntimeException(message)
+  type ItvcResponse[T]  = Either[ErrorCode, T]
+  type ItvcResult  = Future[Result]
+  implicit def toEither[T](t: T): ItvcResponse[T] = Right(t)
```
Comment (avelx) [inline app/controllers/ChargeSummaryController.scala:50]: I would suspect this is something we can cover by using cats library which is in scope of this project

Comment (james-king-hmrc) [reply app/controllers/ChargeSummaryController.scala:50]: indeed, already investigated using Cats and we can use EitherT transformer for this type of conversion but didn't want to spend to much time perfecting it before getting some initial feedback, will do

Comment (james-king-hmrc) [reply app/controllers/ChargeSummaryController.scala]: tried to use Cats, it didn't compile for some reason happy to pair with someone on this, but i don't favour invisible implicits that aren't on the screen for me to see, I prefer implicit type extensions that get called explicitly and are visible on the screen to give a hint to what's going on in the code, happy to discuss :)

File: app/utils/ImplicitsWithFunUtils.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+package utils
+
+import play.api.mvc.Result
+
+import scala.concurrent.Future
+import scala.language.implicitConversions
+
+object ImplicitsWithFunUtils {
```
Comment (avelx) [inline app/utils/ImplicitsWithFunUtils.scala:24]: this should come via proposal stage as this fundamentals different from the way we do in the rest of the code base; the short answer, not in the scope of this PR pls

Comment (james-king-hmrc) [reply app/utils/ImplicitsWithFunUtils.scala:24]: okay no problem, will move code to controller

**Comment by ian-hmrc** [comment]
I'm struggling to review this PR with so much change and large commits. I like the final result, but I can't be sure that it is a refactoring that provides equivalent behaviour to the original code. Would it be possible to break this down into smaller, cleaner commits to show the connection from the original code to the final result, and are easy to review?

**Pull request closed without merging**